pkg/kubelet/kubelet.go

```diff
@@ -2176,6 +2187,9 @@ func (kl *Kubelet) syncLoopIteration(configCh <-chan kubetypes.PodUpdate, handle
 			// TODO: Do we want to support this?
 			glog.Errorf("Kubelet does not support snapshot update")
 		}
+
+		kl.sourcesReady.AddSource(u.Source)
```

Comment:
yuju also created a PR #30417 for this change explicitly and she added good comments about it.